src/v2/Apps/Partner/Components/Overview/ShowsRail.tsx

```diff
@@ -17,44 +17,47 @@ const ShowsRail: React.FC<ShowsRailProps> = ({ partner, ...rest }) => {
   // @ts-expect-error STRICT_NULL_CHECK
   const [isSeeAllAvaliable, setIsSeeAllAvaliable] = useState<boolean>(undefined)
   if (
-    !partner ||
-    !partner.showsConnection ||
-    // @ts-expect-error STRICT_NULL_CHECK
+    !partner?.showsConnection?.edges ||
     partner.showsConnection.edges.length === 0
   ) {
     return null
   }
 
   const {
     slug,
-    showsConnection: { edges: shows },
+    showsConnection: { edges },
+    fullProfileEligible,
   } = partner
 
+  const canShowAll = isSeeAllAvaliable && fullProfileEligible
+  const shows = [...edges]
+
+  if (fullProfileEligible && shows.length === 20) {
+    shows.pop()
+  }
```

Comment:
This is fine, just wondering the "why" of it?

Reply:
well, if we have a non-subscriber then we have to display 20 cards without `view all` card. But if we have a subscriber we have to display 19 cards + `view all` card. I decided that it would be more efficient to replace the last card with `view all` card if a user is subscriber.

The issue was that we have to fetch first 20 for subs and first 19 for non subs.